app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -278,6 +274,13 @@ def infer_related_ems_refs!
       infer_related_cloud_volumes_ems_refs_db!
       infer_related_cloud_volumes_ems_refs_api!
     end
+    unless references(:cloud_tenants).blank?
+      # need to reset the association cache so that tenants added by the
+      # previous infer methods get picked up
+      target.manager_refs_by_association_reset
+      infer_related_cloud_tenant_ems_refs_db!
+      infer_related_cloud_tenant_ems_refs_api!
```

Comment:
this is probably still not enough, since the infer_related_cloud_tenant_ems_refs_api! still ads to ```references(:cloud_tenants)```, so whatever is added by that would not be fetched by the API

i think it might be enough to just cache the API call here? Or do we call the ```tenants``` method often?

Reply:
`infer_related_cloud_tenant_ems_refs_api!` is where `tenants` is used for the first time, so I think I should just need to move `target.manager_refs_by_association_reset` into there.

Reply:
right, but we'll need to also fetch the tenants again, since we are caching them here https://github.com/ManageIQ/manageiq-providers-openstack/blob/ac5abeab9386bb2cdcdfbf8e7d0ba9af0a09fb0e/app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb#L125

Reply:
so we should do both ```target.manager_refs_by_association_reset``` and ```@tenants = nil```, or maybe do a non cached call

Reply:
yes, I think we don't call the ```tenants``` much right? So it should be enough that the API call is cached